hedera-node/src/main/java/com/hedera/services/txns/token/BurnLogic.java

```diff
@@ -0,0 +1,61 @@
+package com.hedera.services.txns.token;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.state.enums.TokenType;
+import com.hedera.services.store.AccountStore;
+import com.hedera.services.store.TypedTokenStore;
+import com.hedera.services.store.models.Id;
+import com.hedera.services.store.models.OwnershipTracker;
+
+import java.util.List;
+
+public class BurnLogic {
```

Comment:
I think that we should make this injectable as MintLogic: https://github.com/hashgraph/hedera-services/pull/2516/files#diff-e50754ffc994d0767f46b936ffe56a9948363bdc519b0b3b7044f24884fd13e1R43